rustls/src/server/server_conn.rs

```diff
@@ -120,13 +121,26 @@ pub trait ResolvesServerCert: Debug + Send + Sync {
     ///
     /// Return `None` to abort the handshake.
     fn resolve(&self, client_hello: ClientHello<'_>) -> Option<Arc<sign::CertifiedKey>>;
+
+    /// Return true when the server only supports raw public keys.
+    fn only_raw_public_keys(&self) -> bool {
+        false
+    }
 }
 
+/// A struct representing the server certificate types offered in the ClientHello.
+pub(super) struct ServerCertType<'a>(pub(super) &'a [CertificateType]);
+
+/// A struct representing the client certificate types offered in the ClientHello.
+pub(super) struct ClientCertType<'a>(pub(super) &'a [CertificateType]);
```

Comment:
I think the visibility of these two items means they should be lower in the mod (below the `pub` items).